relm4-components/src/alert/mod.rs

```diff
@@ -0,0 +1,244 @@
+                        // - On Adwaita, we want the *text* of the button to be red.
+                        #[name(confirm_label)]
+                        gtk::Button {
+                            #[watch]
+                            set_visible: model.settings.confirm_label.is_some(),
+                            #[watch]
+                            set_class_active: (DESTRUCTIVE_CSS, !LIBADWAITA_ENABLED && model.settings.destructive_accept),
+                            #[watch]
+                            set_class_active: (FLAT_CSS, LIBADWAITA_ENABLED || !model.settings.destructive_accept),
+                            set_hexpand: true,
+                            connect_clicked => AlertMsg::Response(AlertResponse::Confirm),
+
+                            gtk::Label {
+                                #[watch]
+                                set_label: model.settings.confirm_label.as_deref().unwrap_or_default(),
+                                #[watch]
+                                set_class_active: (ERROR_CSS, LIBADWAITA_ENABLED && model.settings.destructive_accept),
+                            }
+                        },
+
+                        gtk::Box {
+                            #[watch]
+                            set_visible: model.settings.cancel_label.is_some(),
+
+                            gtk::Separator {},
+
+                            #[name(cancel_label)]
+                            gtk::Button {
+                                #[watch]
+                                set_label: model.settings.cancel_label.as_deref().unwrap_or_default(),
+                                add_css_class: FLAT_CSS,
+                                set_hexpand: true,
+                                connect_clicked => AlertMsg::Response(AlertResponse::Cancel)
+                            }
+                        },
+
+                        gtk::Box {
+                            #[watch]
+                            set_visible: model.settings.option_label.is_some(),
+
+                            gtk::Separator {},
+
+                            #[name(option_label)]
+                            gtk::Button {
+                                #[watch]
+                                set_label: model.settings.option_label.as_deref().unwrap_or_default(),
+                                add_css_class: FLAT_CSS,
+                                set_hexpand: true,
+                                connect_clicked => AlertMsg::Response(AlertResponse::Option)
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    fn init(
+        settings: AlertSettings,
+        root: Self::Root,
+        sender: ComponentSender<Self>,
+    ) -> ComponentParts<Self> {
+        relm4::set_global_css_with_priority(
+            COMPONENT_CSS,
+            gtk::STYLE_PROVIDER_PRIORITY_APPLICATION,
+        );
```

Comment:
I'm not sure how GTK handles this, but ideally we would call this only the first time an `AlertComponent` is created.

Reply:
I'd assume it would handle it the same way as if we override a styling that already exists (like how a custom theme might do). The styling shouldn't need to be initialized more than once in this case anyway though.